modules/core/src/main/java/org/apache/ignite/lang/ByteArray.java

```diff
@@ -22,15 +22,16 @@
 import org.jetbrains.annotations.NotNull;
 
 /**
- * A class for handling byte array.
+ * A class wraps {@code byte[]} which provides {@code equals}, {@code hashCode} and lexicographical comparison.
```

Comment:
Maybe it will be better to use javadoc links to Object#equals and hashCode here.